File: StudyGamePrograming08/AsteroidControl.cpp
#include "AsteroidControl.h"
#include "Game.h"

AsteroidControl::AsteroidControl(Game* game):Actor(game)
    , InitCooldown(5.0f)
    , mAsteroidCooldown(InitCooldown)    
{
}

void AsteroidControl::UpdateActor(float deltaTime)
{
	mAsteroidCooldown -= deltaTime;
    //小惑星を一定時間ごとに増やす。小惑星の数が０になったらゲームクリア画面をアクティブにする。
    int numAsteroids = GetGame()->GetAsteroids().size();
    if(mAsteroidCooldown < 0.0 && numAsteroids > 0)
    {
        GetGame()->AddAsteroid();
        mAsteroidCooldown = InitCooldown;
    }        
}

File: StudyGamePrograming08_VB/AsteroidControl.vb
Public Class AsteroidControl
    Inherits Actor

    Private mAsteroidCooldown As Double      '小惑星増殖までの待機時間
    Private InitCooldown As Double      '小惑星増殖までの待機時間

    Public Sub New(ByRef game As Game)
        MyBase.New(game)
        InitCooldown = 5.0
        mAsteroidCooldown = InitCooldown
    End Sub
    Public Overrides Sub UpdateActor(ByVal deltaTime As Double)
        mAsteroidCooldown -= deltaTime
        '小惑星を一定時間ごとに増やす。小惑星の数が０になったらゲームクリア画面をアクティブにする。
        Dim numAsteroids As Integer = GetGame().GetAsteroids().Count()
        If (mAsteroidCooldown < 0.0 And numAsteroids > 0) Then
            GetGame().AddAsteroid()
            mAsteroidCooldown = InitCooldown
        End If
    End Sub
End Class